mlrun/model_monitoring/writer.py

```diff
@@ -11,27 +11,143 @@
+        self._tsdb_client = self._get_v3io_frames_client()
+        self._create_tsdb_table()
 
-    def __init__(self, name: str = None):
-        self.name = name or "king"
+    def _get_v3io_client(self) -> V3IOClient:
+        return mlrun.utils.v3io_clients.get_v3io_client(endpoint=mlrun.mlconf.v3io_api)
 
-    def do(self, event):
+    def _get_v3io_frames_client(self) -> V3IOFramesClient:
+        return mlrun.utils.v3io_clients.get_frames_client(
+            address=mlrun.mlconf.v3io_framesd,
+            container=self._v3io_container,
+        )
+
+    def _create_tsdb_table(self) -> None:
+        self._tsdb_client.create(
+            backend=_TSDB_BE,
+            table=self._TSDB_TABLE,
+            if_exists=IGNORE,
+            rate=_TSDB_RATE,
+        )
 
-        print(
-            f"endpoint_uid ={event[mlrun.common.schemas.model_monitoring.constants.WriterEvent.ENDPOINT_ID]}, \n"
-            f"app_name = {event[mlrun.common.schemas.model_monitoring.constants.WriterEvent.APPLICATION_NAME]}, \n"
-            f"schedule_time = {event[mlrun.common.schemas.model_monitoring.constants.WriterEvent.SCHEDULE_TIME]}, \n"
-            f"result_name ={event[mlrun.common.schemas.model_monitoring.constants.WriterEvent.RESULT_NAME]}, \n"
-            f"result_value ={event[mlrun.common.schemas.model_monitoring.constants.WriterEvent.RESULT_VALUE]}, \n"
-            f"my_name = {self.name}."
+    def _update_kv_db(self, event: AppResultEvent) -> None:
+        event = AppResultEvent(event.copy())
+        endpoint_id = event.pop(WriterEvent.ENDPOINT_ID)
+        app_name = event.pop(WriterEvent.APPLICATION_NAME)
+        self._kv_client.put(
+            container=self._v3io_container,
+            table_path=endpoint_id,
+            key=app_name,
+            attributes=event,
         )
-        return event
+        logger.info("Updated V3IO KV successfully", key=app_name)
+
+    def _update_tsdb(self, event: AppResultEvent) -> None:
+        event = AppResultEvent(event.copy())
+        event[WriterEvent.SCHEDULE_TIME] = pd.to_datetime(
+            event[WriterEvent.SCHEDULE_TIME],
+            format=EventFieldType.TIME_FORMAT,
+        )
+        try:
+            self._tsdb_client.write(
+                backend=_TSDB_BE,
+                table=self._TSDB_TABLE,
+                dfs=pd.DataFrame.from_records([event]),
+                index_cols=[
+                    WriterEvent.SCHEDULE_TIME,
+                    WriterEvent.ENDPOINT_ID,
+                    WriterEvent.APPLICATION_NAME,
+                ],
+            )
+            logger.info("Updated V3IO TSDB successfully", table=self._TSDB_TABLE)
+        except V3IOFramesError as err:
+            logger.warn(
+                "Could not write drift measures to TSDB",
+                err=err,
+                table=self._TSDB_TABLE,
+                event=event,
+            )
+
+    @staticmethod
+    def _reconstruct_event(event: RawEvent) -> AppResultEvent:
```

Comment:
what is the advantage of that `RawEvent` object which is basically a dictionary? it's not that we are doing here some validations on that object. I think it's more clear just to define `event: Dict[str, Any]`. You can see that it is a very common use across mlrun. btw the same question for `AppResultEvent`.

Reply:
It's just for convenience in the module - perhaps I should signify it with a leading underscore (`RawEvent` -> `_RawEvent`).
`RawEvent` - a raw dictionary.
`AppResultEvent` - after checking all the keys.